From 62030d589336fc052eab8cf4503ef1968f5d9ac8 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@php.net>
Date: Thu, 25 Sep 2014 10:16:51 +0200
Subject: [PATCH] Fix build with curl < 7.26

$ grep  "define CURL_REDIR_POST_30" curl-7.2*/include/curl/curl.h

curl-7.25.0/include/curl/curl.h:#define CURL_REDIR_POST_301 1
curl-7.25.0/include/curl/curl.h:#define CURL_REDIR_POST_302 2

curl-7.26.0/include/curl/curl.h:#define CURL_REDIR_POST_301 1
curl-7.26.0/include/curl/curl.h:#define CURL_REDIR_POST_302 2
curl-7.26.0/include/curl/curl.h:#define CURL_REDIR_POST_303 4
---
 php_http_client_curl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/php_http_client_curl.c b/php_http_client_curl.c
index 6f9884e..8f25211 100644
--- a/php_http_client_curl.c
+++ b/php_http_client_curl.c
@@ -2169,7 +2169,9 @@ PHP_MINIT_FUNCTION(http_client_curl)
 #if PHP_HTTP_CURL_VERSION(7,19,1)
 	REGISTER_NS_LONG_CONSTANT("http\\Client\\Curl", "POSTREDIR_301", CURL_REDIR_POST_301, CONST_CS|CONST_PERSISTENT);
 	REGISTER_NS_LONG_CONSTANT("http\\Client\\Curl", "POSTREDIR_302", CURL_REDIR_POST_302, CONST_CS|CONST_PERSISTENT);
+#if PHP_HTTP_CURL_VERSION(7,26,0)
 	REGISTER_NS_LONG_CONSTANT("http\\Client\\Curl", "POSTREDIR_303", CURL_REDIR_POST_303, CONST_CS|CONST_PERSISTENT);
+#endif
 	REGISTER_NS_LONG_CONSTANT("http\\Client\\Curl", "POSTREDIR_ALL", CURL_REDIR_POST_ALL, CONST_CS|CONST_PERSISTENT);
 #endif
 
-- 
2.1.0

